### PR TITLE
Add CollectionAge validation to GrainCollectionOptions

### PR DESCRIPTION
**Pull request by xiazen**
Fix issue #5200

### CODE REVIEW

File: src/Orleans.Runtime/Configuration/Validators/CollectionAgeLimitValidator.cs [hunk 1]
```diff
@@ -4,17 +4,24 @@
...
             this.options = options;
         }
 
         public void ValidateConfiguration()
         {
+            if (this.options.Value.CollectionAge <= this.options.Value.CollectionQuantum)
+            {
+                throw new OrleansConfigurationException(
+                    $"{nameof(GrainCollectionOptions.CollectionAge)} is set to {options.Value.CollectionAge}. " +
+                    $"{nameof(GrainCollectionOptions.CollectionAge)} must be bigger than {nameof(GrainCollectionOptions.CollectionQuantum)}, " +
```
Comment (ReubenBond) [inline src/Orleans.Runtime/Configuration/Validators/CollectionAgeLimitValidator.cs]: nit: greater than

**Comment by xiazen** [comment]
@dotnet-bot test functional please

**Comment by ReubenBond** [comment]
Test issue:
> `Orleans.Runtime.OrleansConfigurationException : UnitTests.Grains.CollectionSpecificAgeLimitForTenSecondsActivationGcTestGrain CollectionAgeLimit is set to 00:00:12. CollectionAgeLimit must be greater than CollectionQuantum, which is set to 00:01:00`

**Comment by sergeybykov** [comment]
@dotnet-bot test functional

**Comment by sergeybykov** [comment]
[Functional test failures](https://ci.dot.net/job/dotnet_orleans/job/master/job/functional_prtest/2498/):
>UnitTests.TimeoutTests.Timeout_LongMethod
UnitTests.ActivationsLifeCycleTests.ActivationCollectorTests.ActivationCollectorShouldCollectIdleActivationsSpecifiedInPerTypeConfiguration
UnitTests.ActivationsLifeCycleTests.ActivationCollectorTests.ActivationCollectorShouldCollectByCollectionSpecificAgeLimitForTenSeconds
UnitTests.ActivationsLifeCycleTests.ActivationCollectorTests.ActivationCollectorShouldNotCollectBusyActivationsSpecifiedInPerTypeConfiguration
UnitTests.ActivationsLifeCycleTests.ActivationCollectorTests.ActivationCollectorShouldNotCollectIdleActivationsIfDisabled

**Pull request merged**